### ISSUE TITLE
758 resource identifier

### ISSUE THREAD

**Issue by CECSpecialistI** [uwlib-cams/MARC2RDA#268]
https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/7XX.csv

**Comment by cwarc** [comment]
https://docs.google.com/spreadsheets/d/1_E3wrUKGZaNpkPkPyylp2JFLnq8hNjzwunpGHFb2GL0/edit#gid=523701703

Hello Everyone,

Please see the first pass for field code 758  Resource Identifier (R). Could anyone take a look?

Thanks.

Best,
Alice

**Comment by cspayne** [comment]
Mapping $4 and $1 seems pretty straightforward: if $4 is an RDA WEMI property, map it as such with the $1 URI as the object. 
$a and $0 seem trickier, because they are nomens associated with the object identified in $1. I don't see a way of determining what type of RDA entity that object is, which is how we would determine what property to use when mapping $a and $0.

**Comment by GordonDunsire** [comment]
@cspayne and colleagues: I agree that there is a straightforward transform for some of the instance data, based on the use of an RDA property.

If we assume that the property value is correct and valid, then the condition for an RDA property can be simplified to a generic function: position(subfield, 'rdaregistry.info') > 0. This tests if the IRI contains 'rdaregistry.info'. Anything more elaborate is a form of validation.

Complications :-(

1. 758 subfield $4 says: 'Code or URI that specifies the relationship from the entity described in the record to the entity referenced in the field.'

This implies that only one entity is described in 'the record'. (Unless I am misinterpreting US English :-)

Subfield $i says '... the resource described in field 758 and the resource described in the 1XX/245 of the record'.

This indicates that the 'resource' or entity is determined by 1XX/245.

In the example for subfield $a, there are two occurrences of the field, one for a manifestation and one for a work. This appears to contradict the implication that only one entity is being described.

The complication for the transform is:

Which entity is the subject of the 'record'/field 758/transformed statement:

<thisEntity> <rda relationship property> <relatedEntity> .

Is <thisEntity> 'the' instance of manifestation being described, or 'the' instance of work that is embodied by the manifestation? This is further complicated when whole-part manifestations and aggregates (multiple embodied works) are being described.

2. The domain of the RDA property recorded in subfield 4 is the class of entity that is the subject of the statement. The range of the property is the class of the related entity.

This is a strong declaration. If we assume that the data is valid, I think that this should have a high priority if there is a collision with the transform of other fields. That is, if there is doubt about the classes of instances that are generated by the transform overall, field 758 should be a deciding factor. I am thinking particularly of expressions rather the the work/manifestation associations implied by the manual.

3. Subfield $0 can be mapped to an RDA identifier recording method for the property that is used. So the same straightforward transform can include either or both of subfields $0 and $1.

If the RDA property is a canonical property, it can be transformed to a datatype property (for subfield $0) or an object property (for subfield $1) using a simple process:

If exists($0) replace($4, '/P', '/datatype/P')
If exists($1) replace($4, '/P', '/object/P')

[The Registry and namespaces were designed for this purpose.]

4. If an RDA property is not used, then we may be able to develop an approach based on the work with relationship properties in other 7XX fields. That is, use a mapping from the other property to an RDA property.

Uncomplications :-)

Subfield $a seems straightforward. We know that it is a label for the related entity, but we don't know what kind and there is no way of automatically determining if a string is an authorized access point or title, etc. So the transform should use a general RDA appellation property for the entity of subfield $1:

<$1> <RDA appellation of entity> "$a" .

In the MARC manual example:

http://www.wikidata.org/entity/Q1613936 rdamd:P30277 "Bored of the Rings" . // property determined by the range of $4
http://id.loc.gov/authorities/names/no97079452 rdawd:P10329 "Tolkien, J. R. R. (John Ronald Reuel), 1892-1973. Lord of the rings" . // property determined by the range of $4

Is this in scope? Should we add RDA properties to existing instances, or leave them encapsulated for de-referencing? Similar (non-RDA) properties are attached to the existing instances ("label" in wikidata, the aap in LC authorities).

It is clearly in scope for instances that are minted in the transform, but such minting is triggered by other fields so is not directly relevant to this field.

**Comment by lake44me** [comment]
At the BIBFRAME Update Forum Jan. 22, Jeff Mixter of OCLC said they would start adding WorldCat Entity URIs to Worldcat records at the end of January (so presumably they are going in now) for Works.  I asked in the Q/A what the WorldCat definition for "Work" was, and was told it aligned with BIBFRAME Work.  

Since various people at LC have stated that a BIBFRAME Work is equivalent to RDA Expression with some RDA Work characteristics included (I'd have to hunt down references to this in print...), would this support taking the position that the entity being related most closely corresponds to an RDA Expression, and that what it is related to is the RDA Manifestation whose description is the focus of the record?  

Jeff said they would use a 758 field but didn't show an example.  While we can't assume everyone using 758 will follow OCLC's pattern in that use, as soon as this is documented, we should analyze it and the property asserted.

@GordonDunsire hate to say it but your use of the word instance befuddles me because I'm entangled with BIBFRAME now for another project and bf:instance is a thing.  Hard to find words that haven't been used elsewhere with a different meaning these days!

**Comment by GordonDunsire** [comment]
@lake44me: I use the word 'instance' in its RDF sense, which is based on the vernacular use as 'an example or single occurrence of something': a single occurrence of a thing (i.e. of a class). RDF Schema (RDFS), since 2004, uses the term frequently and consistently: 'The members of a class are known as instances of the class'. I think this usage predates that of BIBFRAME. BF could easily have used the term "manifestation", which is used elsewhere with the same meaning, but chose to use a befuddling term instead. I don't speak BF at all, but I know a little RDF/LRM and I think that's more appropriate for an RDA project :-)

**Comment by cspayne** [comment]
@CECSpecialistI Could we see if someone is willing to take this mapping from me and complete it? I'm afraid I won't have time to.

**Comment by GordonDunsire** [comment]
@cspayne: What needs to be done? I can analyze further the field from an RDA pov, in light of discussions on headings, etc. since February. Does this Issue thread constitute the documentation on what has already been done (that needs to be completed) or is there relevant information elsewhere?

**Comment by cspayne** [comment]
@GordonDunsire 
Yes, all discussion that was done is in this issue. We need to decide how we want to map each of these subfields and fill out the Google Sheet so that it can be moved to ready for review.

**Comment by GordonDunsire** [comment]
@cspayne and others: I don't want to get entangled in the Google Sheet, but here's my further analysis:

$a is a nomen [string] for a resource entity (WEMI).
$0 is an identifier for the same resource entity.
$1 is the same resource entity.
$2 is controlled and indicates the authority of the nomen for the resource entity.

$i is uncontrolled.
$4 is controlled. RDA properties are easily identifiable.

$3 and $5 are difficult to map to RDA.

Further discussion:

Mapping of 1xx/245 to RDA WEM entities has been resolved in the context of phase 1. A work and a manifestation will always be minted, so there is always more than one entity identified by the 1xx/245 values. The complications of aggregates and diachronc resources are only relevant as context and future development.

The RDA entity that is the domain of the RDA entity in $4 is a priority in determining the entities transformed from 1xx/245.

Mapping of $a, $0, $1, $2 has been resolved. This includes approval of $1 sources to avoid range entity-collision with $4. (This also supersedes my note 3 above.)

Therefore the transform is primarily dependent on whether $4 is an RDA property. If it is not, then we need to determine a default.

The cases of an unconstrained RDA property, and an original and official RDA label in $i, are dealt with elsewhere (I'm not sure of the current status of the relationship labels work).

Transform:

If either of $3, $5 are present, drop the field.

If $4 (string) contains 'rdaregistry.info' and does not contain 'rdaregistry.info/Elements/u' then:

- mint an entity for the domain of the property or re-use *** the appropriate work or expression minted by 1xx/245
- select the entity or value for the range of the property. Process $a, $0, $1, and $3 the same as for heading fields.
- output the statement (domain entity) objectified($4) (minted range entity) or (domain entity) datatyped($4) 'range string' as appropriate. (The datatyped and objectified functions generate the datatype and object versions of the RDA elements.)
- Ignore $i.

If $4 (string) does not contain 'rdaregistry.info' or contains 'rdaregistry.info/Elements/u' then output a note on manifestation with the contents of the field as a structured string. There is no easy way of determining the domain or range in phase 1.

If $4 is missing, transform $i as for other 7xx fields.

*** I'm sill wondering if we can de-duplicate the primary WE that are minted for each resource. The big problems are augmentation aggregates, which are likely to have the same 1xx/245 values as a non-aggregate, and the lack of an authority source. In this case, however, we could make the assumption that the WE referenced in field 758 are the same as the WE minted from 1xx/245, if aggregates are excluded. That is, if an RDA Work or Expression element is used in $4, then use the instance of W or E that is minted from 1xx/245.

**Comment by cspayne** [comment]
@CECSpecialistI @sarahruska Gordon has done a lot of the mental work for this mapping but someone needs to go in and do the mapping spreadsheet

**Comment by CECSpecialistI** [comment]
@corialanus are you up for doing the mapping spreadsheet for this one? Gordon has done a lot of thought on it already.

**Comment by corialanus** [comment]
@CECSpecialistI  yes I can take this. It is also a MARC field we are using at NLNZ for encoding Manifestation: expression manifested

**Comment by corialanus** [comment]
I am not confident that I have mapped this well from Gordon's analysis so this needs very careful review.

**Comment by tmqdeborah** [comment]
758 can be used to tell us: “either the resource described in the bibliographic record or a resource to which it is related”.
Questions:
- What work is described by the record (UW) or expression (NLNZ)?
  - The same as we figure out from the 1XX/2XX (and 7XXInd2=2?)?
- Why provide it
  - To indicate the primary relationship for the entity described by the 1XX/2XX (and 7XXInd2=2?), using $i (not available for 240)
 - Related WEM 
   - Instead of using other 7XX (UW)?
BF conversion specs is a single instruction for 758: if $1 exists, create secondary Instance(s) to existing Work ; add rdf:resource="http://id.loc.gov/ontologies/bflc/SecondaryInstance"
<skos:definition>
Instance created when a single MARC record is split into multiple MARC records for conversion to BIBFRAME; used to connect multiple Instance resources to a single Work

It is very confusing about how this field is used by any individual library. I recommend postponing until Phase II

**Comment by CECSpecialistI** [comment]
Moved to Phase II per Deborah's assessment
